packages/atti-components/src/components/Slider/index.js

```diff
@@ -0,0 +1,104 @@
+  const [indicatorPos, setIndicatorPos] = useState(0)
+  const [xMousePosition, setXMousePosition] = useState(0)
+
+  const pxSliderSize = sliderRail.sizes.width.slice(0, -2)
+  // Add event regarding the indicator motion, remove it when Slider is unmounted
+  useEffect(() => {
+    window.addEventListener('mousemove', moveIndicatorEvent)
+    window.addEventListener('mouseup', onMouseUp)
+
+    return () => {
+      window.removeEventListener('mousemove', moveIndicatorEvent)
+      window.addEventListener('mouseup', onMouseUp)
+    }
+  })
+
+  const onMouseUp = () => {
+    if (isPressed) setIsPressed(false)
+  }
+
+  const onMouseDown = e => {
+    setIsPressed(true)
+    // Use the current mouse position on click,
+    // minus the current slider position so it does not start from 0
+    setXMousePosition(e.clientX - indicatorPos)
+  }
+  // Calculate how much the mouse has moved on X axis relatively to the slider width
+  const calculateIndicatorPos = (currentXMousePosition) => {
+    const newPosition = max([min([currentXMousePosition - xMousePosition, pxSliderSize]), 0])
+    return newPosition
+  }
+
+  const renderSteps = () => map(steps, step =>
+      // Avoid display of 0 and 100 steps
+    !(step.position === 0 || step.position === 100) && <SliderStep key={step.position} leftPosition={step.position}>
+      <SmallText>{step.title}</SmallText>
+    </SliderStep>
+  )
+  // Use the range provided in props to adapt the value returned
+  const getSliderValueInPercent = () => range[0] + round((indicatorPos / pxSliderSize) * (range[1] - range[0]), precision)
+
+  const moveIndicatorEvent = e => {
+    if (!isPressed) return
+    let newIndicatorPos = calculateIndicatorPos(e.clientX)
+    // If values are restricted to steps, find the closest one
+    if (steps && isStepRestricted) {
+      // Calculate the position from the closest step
+      // to the indicator value if it was not restricted
+      newIndicatorPos = minBy(steps, (step) => {
+        const mousePos = newIndicatorPos * 100 / pxSliderSize
+        return Math.abs(mousePos - step.position)
+      }).position * pxSliderSize / 100
+    }
+    setIndicatorPos(newIndicatorPos)
+    // Use onValue changed handler passed as a prop
+    onValueChange && onValueChange(getSliderValueInPercent())
+  }
+
+  return <SliderRail>
+      <SliderIndicator
+        onMouseDown={onMouseDown}
+        leftPosition={indicatorPos}
+        moving={isPressed}
+      >
+      {isValueDisplayed && <SliderValue>{getSliderValueInPercent()}</SliderValue>}
+      </SliderIndicator>
+      {steps && renderSteps()}
+  </SliderRail>
+}
+
+Slider.defaultProps = {
+  isStepRestricted: false,
+  range: [0, 100],
+  precision: 0,
+}
+
+Slider.propTypes = {
+  isStepRestricted: PropTypes.boolean,
+  onValueChange: PropTypes.func,
+  // The value precision to round down to, 1 is corresponding to 0.1, 0 to 1, -1 to 10 and so on
+  precision: PropTypes.oneOf([2, 1, 0, -1, -2, -3, -4, -5, -6]),
```

Comment:
This seems a bit odd....I don't get your comment neither why we have this exhaustive list of numbers in particular

Reply:
The round lodash function is using a integer representing a power of 10. This list is providing limited precision so the end user will not get a returned value like '10.9563'. Has the calcula is done with 'px' values below than '-2' is not really accurate.

Reply:
I feel like this is too resctricting....and not really robust. I think `PropTypes.number` gives enough boundaries and make our calculations scalable

Reply:
Done replacing precision by a step prop easier to handle.